lib/upgrade/migrator/migrate.go

```diff
@@ -676,18 +675,25 @@ func (m *migrationHelper) queryAndConvertV1ToV3Nodes(data *MigrationData) error
 // If neither the v1 nor v3 versions are present, then no migration is required.
 func (m *migrationHelper) ShouldMigrate() (bool, error) {
 	ci, err := m.clientv3.ClusterInformation().Get(context.Background(), "default", options.GetOptions{})
-	if err == nil {
+    if err == nil && ci.Spec.CalicoVersion != ""{
```

Comment:
I did the logic this way to reduce the complexity of the diff.  I coulda done:

```
if err == nil {
  if CalicoVersion != "" {
    ... check if requires migration ...
  }
  ... treat as non-existent ...
}
```

but everything is shifted over and the diff is harder to read.

Can change though if you'd prefer.